casper/protocols/concurrent/concurrent_view.py

```diff
@@ -0,0 +1,67 @@
+"""The blockchain view module extends a view for blockchain data structures """
+import random as r
+
+from casper.safety_oracles.clique_oracle import CliqueOracle
+from casper.abstract_view import AbstractView
+import casper.protocols.concurrent.forkchoice as forkchoice
+
+
+class ConcurrentView(AbstractView):
+    """A view class that also keeps track of a last_finalized_estimate and children"""
+    def __init__(self, messages=None, genesis_block=None):
+        self.children = dict()
+        self.last_finalized_estimate = set([genesis_block])
+        self.genesis_block = genesis_block
+
+        self._initialize_message_caches(messages)
+
+        super().__init__(messages)
+
+    def estimate(self):
+        """Returns the current forkchoice in this view"""
+        available_outputs, output_dict = forkchoice.get_fork_choice(
+            self.last_finalized_estimate,
+            self.children,
+            self.latest_messages
+        )
+
+        num_outputs = r.randint(1, len(available_outputs))
```

Comment:
Probably put this behind a method with a comment on what's going on here. Will allow it to be easy to swap out a different way to consume outputs and create new inputs in the future.

Reply:
`select_outputs_to_consume(available_outputs)`
`create_new_outputs(num)`